www/js/lib/zimArchive.js

```diff
@@ -190,12 +231,16 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8'],
         }, true).then(function(firstIndex) {
             var dirEntries = [];
             var addDirEntries = function(index) {
-                if (index >= firstIndex + resultSize || index >= that._file.articleCount)
+                if (search.state === 'cancelled' || index >= firstIndex + resultSize || index >= that._file.articleCount)
```

Comment:
To be able to match a search that has been replaced by another one, we could replace this line by :
`                if (search.state === 'cancelled' || search.prefix !== prefix || index >= firstIndex + resultSize || index >= that._file.articleCount)`

(and `prefix` might be renamed to `originalPrefix` or whatever same parameter name used in other functions)